### PR TITLE
PIE-3778 Make the Notice icon alignment adjust with the Notice content

### PR DESCRIPTION
**Pull request by andrea-sdl**
## Description

This PR is the third iteration on improving how we align the NoticeIcon in the Notice. 
Our goal is to have the notice align between the first and second line of the content and, if the content is short, be vertically centered.

Context:
In #277 we settled for having two options
1. If we don't have a notice title, we'll vertical align it in the middle.
2. If we have the title, we'll align between the first and second line of text.

Sadly, one might want to add a specific title in the content of the notice and this would show the icon in the middle.

Example: 
CleanShot 2023-11-06 at 13 06 31@2x

As such, I decided to see if Flex could help us build a way to have the vertical alignment adjust properly.

Long story short: I was able to 100% match our goals by using an empty div that will stretch if the height is enough (aligning the icon between the first and the second line) and disappear with short content.

Thanks to that, now the icon should properly align.

Result:
CleanShot 2023-11-06 at 17 38 16@2x

## Checklist

- [ ] This PR has good automated test coverage
- [ ] The storybook for the component has been updated

## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Pull down PR.
1. `npm run dev`.
1. Check that the notice icon properly aligns

### CODE REVIEW

**Review by kat3samsin — APPROVED** [review]
Thank you for your efforts improving this! 🥇

**Comment by djalmaaraujo** [comment]
@andrea-sdl Can you fix the alignment of the icon vertically in some situations? See the screenshot: 

image

Is this how the Figma is expecting? If so, let me know, I am good!

**Comment by andrea-sdl** [comment]
No, that's not correct. Thanks @djalmaaraujo for pointing that out!

**Comment by andrea-sdl** [comment]
@djalmaaraujo I updated the story + the code, mind trying again?

**Review by djalmaaraujo — APPROVED** [review]
